pkg/actor/director.go

```diff
@@ -352,3 +360,50 @@ func (cd *clusterDirector) needsInitialization(cluster *resource.Cluster) bool {
 	}
 	return true
 }
+
+func (cd *clusterDirector) needsIngress(ctx context.Context, cluster *resource.Cluster) (bool, error) {
+	conditions := cluster.Status().Conditions
+	conditionInitializedTrue := condition.True(api.CrdbInitializedCondition, conditions)
+	ingressConditionTrue := condition.True(api.CrdbIngressExposedCondition, conditions)
+
+	// In order to expose ingress,
+	// - the cluster initialized condition must be true
+	// - if there is a change in ingress resource
+
+	if !conditionInitializedTrue {
+		return false, nil
+	}
+
+	// this is the case of update when ingress is removed from CR
```

Comment:
nit: maybe name the function `needsIngressDeploy`? it's a little confusing in this case when the name is `needsIngress` but we need to remove the ingress